ApplicationLibCode/ProjectDataModel/WellLog/RimWellLog.cpp

```diff
@@ -0,0 +1,101 @@
+/////////////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (C) 2024-     Equinor ASA
+//
+//  ResInsight is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  ResInsight is distributed in the hope that it will be useful, but WITHOUT ANY
+//  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+//  FITNESS FOR A PARTICULAR PURPOSE.
+//
+//  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+//  for more details.
+//
+/////////////////////////////////////////////////////////////////////////////////
+
+#include "RimWellLog.h"
+
+#include "RimFileWellPath.h"
+#include "RimTools.h"
+#include "RimWellLogChannel.h"
+
+#include "RiaFieldHandleTools.h"
+#include "RiaQDateTimeTools.h"
+
+#include "cafPdmUiDateEditor.h"
+
+#include <QString>
+
+CAF_PDM_ABSTRACT_SOURCE_INIT( RimWellLog, "WellLog" );
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+const QDateTime RimWellLog::DEFAULT_DATE_TIME = RiaQDateTimeTools::createUtcDateTime( QDate( 1900, 1, 1 ) );
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RimWellLog::RimWellLog()
+{
+    CAF_PDM_InitObject( "Well File Info", ":/LasFile16x16.png" );
+
+    CAF_PDM_InitFieldNoDefault( &m_date, "Date", "Date" );
+
+    CAF_PDM_InitFieldNoDefault( &m_wellLogChannelNames, "WellLogFileChannels", "" );
+    RiaFieldHandleTools::disableWriteAndSetFieldHidden( &m_wellLogChannelNames );
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RimWellLog::~RimWellLog()
+{
+    m_wellLogChannelNames.deleteChildren();
```

Comment:
`m_wellLogChannelNames.deleteChildren();` can be removed. `deleteChildren` will be called in the destructor of the child array field.

Reply:
Fixed.